File: ui/src/lib/types/stats_demo.ts
import { API } from "./api";

export interface StatsDemo {
  map: string;
  roundsCt: number;
  roundsT: number;
}

export const convertStatsDemo = (s: API.StatsDemo): StatsDemo => {
  return {
    map: s.map,
    roundsCt: s.rounds_ct,
    roundsT: s.rounds_t,
  }
}

File: internal/server/dto/stats_demo.go
package dto

import "github.com/topvennie/fragtape/internal/database/model"

type StatsDemo struct {
	Map      string `json:"map"`
	RoundsCT int    `json:"rounds_ct"`
	RoundsT  int    `json:"rounds_t"`
}

func StatsDemoDTO(s *model.StatsDemo) StatsDemo {
	return StatsDemo{
		Map:      s.Map,
		RoundsCT: s.RoundsCT,
		RoundsT:  s.RoundsT,
	}
}
